src/honey_badger/epoch_state.rs

```diff
@@ -307,7 +323,11 @@ where
             } = self.subset_handler.handle(cs_output);
 
             for (k, v) in contributions {
-                step.extend(self.send_decryption_share(k.clone(), &v)?);
+                step.extend(if self.require_decryption {
+                    self.send_decryption_share(k.clone(), &v)?
+                } else {
+                    self.process_decryption(k.clone(), td::Step::default().with_output(v))?
```

Comment:
Let's just directly

```rust
self.decryption.insert(k.clone(), DecryptionState::Complete(v));
```

instead of instantiating a threshold decryption `Step`.